rstudio/rstudio
# implement button for toggling between R / Python

### Intent

Addresses https://github.com/rstudio/rstudio/issues/11613.

### Approach

- Make the programming language logo in the Console Pane clickable (effectively as a menu button), and display a menu below on click.
- Use the existing console language tracker to adapt the console to the language selected.

Since the only transitions are R => Python and Python => R, each transition is controlled by emitting `reticulate::repl_python()` for R => Python, and `quit` for Python => R.

<img width="78" alt="Screenshot 2024-06-07 at 3 11 57 PM" src="https://github.com/rstudio/rstudio/assets/1976582/a90e1d51-05dc-4ac9-9dae-e6866e825ea3">

### Automated Tests

TODO

### QA Notes

Confirm that:

- Clicking R when R is already is active is a no-op, and vice-versa for Python;
- Confirm that clicking the alternate language will properly switch to that language.

### Documentation

N/A

### Checklist
